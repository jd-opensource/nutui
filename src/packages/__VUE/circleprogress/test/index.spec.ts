import { mount } from '@vue/test-utils';
import Circleprogress from '../index.vue';

test('should render component  when using require prop', async () => {
  // 默认圆弧半径为50  2πR为314  默认圆弧宽度为10 所以整个元素宽高为(10+50)*2
  const wrapper = mount(Circleprogress, {
    props: {
      progress: 30
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('should change stoke when use width props', async () => {
  // 圆弧宽度为20 所以整个元素宽高为(20+60)*2
  const wrapper = mount(Circleprogress, {
    props: {
      progress: 40,
      strokeInnerWidth: 20,
      progressOption: {
        radius: 60,
        strokeOutWidth: 20
      }
    }
  });
  let element = wrapper.element as HTMLElement;
  expect(element.style.width).toEqual('160px');
  expect(element.style.height).toEqual('160px');
});
test('should change color when use color props', async () => {
  // 圆弧宽度为20 所以整个元素宽高为(20+60)*2
  const wrapper = mount(Circleprogress, {
    props: {
      progress: 40,
      strokeInnerWidth: 20,
      progressOption: {
        backColor: 'blue',
        progressColor: 'yelllow'
      }
    }
  });
  let circle = wrapper.findAll('circle');
  expect(circle[0].html()).toContain('blue');
  expect(circle[1].html()).toContain('yelllow');
});
