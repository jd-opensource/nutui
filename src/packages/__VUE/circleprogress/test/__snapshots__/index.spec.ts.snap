// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component  when using require prop 1`] = `
"<div class=\\"nut-circleprogress\\" style=\\"height: 120px; width: 120px;\\"><svg height=\\"120\\" width=\\"120\\" x-mlns=\\"http://www.w3.org/200/svg\\">
    <circle r=\\"50\\" cx=\\"60\\" cy=\\"60\\" stroke=\\"#d9d9d9\\" stroke-width=\\"10\\" fill=\\"none\\"></circle>
    <circle r=\\"50\\" cx=\\"60\\" cy=\\"60\\" stroke=\\"red\\" stroke-dasharray=\\"94.2,314\\" stroke-width=\\"10\\" fill=\\"none\\" transform=\\"rotate(-90,60,60)\\" stroke-linecap=\\"round\\" style=\\"transition: stroke-dasharray 0.6s ease 0s, stroke 0.6s ease 0s;\\"></circle>
  </svg>
  <div class=\\"nut-circleprogress-content\\">30%</div>
</div>"
`;
